### PR TITLE
MGMT-14582: Set OCI platform behind a capability

### PR DESCRIPTION
**Pull request by adriengentil**
Ensure OCI platform cannot be set if user's organization doesn't
have `bare_metal_installer_platform_oci` capability.

### CODE REVIEW

File: internal/cluster/validations/validations.go [hunk 1]
```diff
@@ -976,3 +978,30 @@ func ValidateArchitectureWithPlatform(architecture *string, platform *models.Pla
...
+		return nil
+	}
+
+	available, err := authzHandler.HasOrgBasedCapability(ctx, *capabilityName)
+	if err == nil && available {
+		return nil
+	}
+
+	if err != nil {
+		log.WithError(err).Errorf("error getting user %s capability", *capabilityName)
```
Comment (eliorerz) [inline internal/cluster/validations/validations.go:1003]: Why is that log not inside `HasOrgBasedCapability`? I saw that it's duplicated 3 times now

**Review by eliorerz — CHANGES_REQUESTED** [review]

File: internal/cluster/validations/validations.go (hunk 1 above)
Comment (adriengentil) [reply internal/cluster/validations/validations.go]: Done!

**Comment by gamli75** [comment]
@adriengentil there is a fix for it https://github.com/openshift/assisted-service/pull/5259
rebase the PR

**Pull request merged**